docs/decorator-reference.md

```diff
@@ -374,6 +375,36 @@ export class User {
 
 Value will be generated only once, before inserting the entity into the database.
 
+#### `@VirtualColumn`
+
+Special column that is never saved to the database and thus acts as a readonly property.
+Each time you call `find` or `findOne` from the entity manager, the value is recalculated based on the query function that was provided in the VirtualColumn Decorator. The alias argument passed to the query references the exact entity alias of the generated query behind the scenes.
+
+```typescript
+@Entity({ name: "companies", alias: "COMP" })
+export class Company extends BaseEntity {
+  @PrimaryColumn("varchar", { length: 50 })
+  name: string;
+
+  @VirtualColumn({ query: (alias) => `SELECT COUNT("name") FROM "employees" WHERE "companyName" = ${alias}.name` })
+  totalEmployeesCount: number;
+
+  @OneToMany((type) => Employee, (employee) => employee.company)
+  employees: Employee[];
+}
+
+@Entity({ name: "employees" })
+export class Employee extends BaseEntity {
+  @PrimaryColumn("varchar", { length: 50 })
+  name: string;
+
+  @ManyToOne((type) => Company, (company) => company.employees)
+  company: Company;
+}
+```
+
+Value will be generated only once, before inserting the entity into the database.
```

Comment:
Is it generated or it's read-only additional property returned by `find` methods?

Reply:
Whoops - good catch - this should not be in here. This is not inserting into the database. This was a copy/paste mistake from the decorator above.

Reply:
This has been fixed @Ginden